modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/trait/TraitUtils.java

```diff
@@ -175,9 +175,15 @@ else if (converter == RewindabilityTraitDef.INSTANCE)
             return rel;
 
         RelTraitSet traits = rel.getTraitSet()
-            .replace(toTrait);
-
-        return new IgniteTableSpool(rel.getCluster(), traits, Spool.Type.LAZY, rel);
+            .replace(toTrait)
+            .replace(CorrelationTrait.UNCORRELATED);
```

Comment:
Do we need this additional UNCORRELATED here?

Reply:
Correlation is never required by table spool itself, so it's better to remove trait